plugins/runner/runner.go

```diff
@@ -35,35 +34,39 @@ import (
 const defaultPollInterval = 2 * time.Second
 
 type pluginRunner struct {
-	config           types.PluginConfig
-	containerManager containermanager.PluginContainerManager
-	client           runnerclient.ClientWithResponsesInterface
+	config         types.PluginConfig
+	runtimeHandler runtimehandler.PluginRuntimeHandler
+	client         runnerclient.ClientWithResponsesInterface
 }
 
 func New(ctx context.Context, config types.PluginConfig) (types.PluginRunner, error) {
 	// Create docker container
-	// TODO: switch to factory once the support for more container engines is added
-	manager, err := docker.New(ctx, config)
+	handler, err := getPluginRuntimeHandler(ctx, config)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create plugin manager: %w", err)
+		return nil, fmt.Errorf("failed to create plugin runtime handler: %w", err)
 	}
 
 	return &pluginRunner{
-		config:           config,
-		containerManager: manager,
+		config:         config,
+		runtimeHandler: handler,
 	}, nil
 }
 
 func (r *pluginRunner) Start(ctx context.Context) error {
-	if err := r.containerManager.Start(ctx); err != nil {
+	go func(ctx context.Context) {
+		<-ctx.Done()
+		r.Remove(ctx) //nolint:errcheck
```

Comment:
this will call remove multiple times. for docker, its okay, not sure if its okay for binary mode

Reply:
I don't think it is okay, we should track the internal lifecycle of the plugin execution.

Reply:
marking resolved as we can call remove multiple times per discussion